File: core/src/main/scala/com/github/propi/rdfrules/data/TripleItemType.scala
package com.github.propi.rdfrules.data

import scala.language.implicitConversions

/**
  * Created by Vaclav Zeman on 2. 2. 2018.
  */
sealed trait TripleItemType

object TripleItemType {

  case object Resource extends TripleItemType

  case object Number extends TripleItemType

  case object Boolean extends TripleItemType

  case object Text extends TripleItemType

  case object Interval extends TripleItemType

  implicit def apply(tripleItem: TripleItem): TripleItemType = tripleItem match {
    case _: TripleItem.Uri => Resource
    case _: TripleItem.Text => Text
    case _: TripleItem.BooleanValue => Boolean
    case _: TripleItem.Number[_] => Number
    case _: TripleItem.Interval => Interval
  }

}

File: java/src/main/java/com/github/propi/rdfrules/java/TripleItemType.java
package com.github.propi.rdfrules.java;

/**
 * Created by Vaclav Zeman on 10. 5. 2018.
 */
public enum TripleItemType {
    RESOURCE(com.github.propi.rdfrules.data.TripleItemType.Resource$.MODULE$),
    NUMBER(com.github.propi.rdfrules.data.TripleItemType.Number$.MODULE$),
    BOOLEAN(com.github.propi.rdfrules.data.TripleItemType.Boolean$.MODULE$),
    TEXT(com.github.propi.rdfrules.data.TripleItemType.Text$.MODULE$),
    INTERVAL(com.github.propi.rdfrules.data.TripleItemType.Interval$.MODULE$);

    final private com.github.propi.rdfrules.data.TripleItemType tripleItemType;

    TripleItemType(com.github.propi.rdfrules.data.TripleItemType tripleItemType) {
        this.tripleItemType = tripleItemType;
    }

    public com.github.propi.rdfrules.data.TripleItemType getTripleItemType() {
        return tripleItemType;
    }

    public static TripleItemType tripleItemTypeToEnum(com.github.propi.rdfrules.data.TripleItemType tripleItemType) {
        for (TripleItemType value : TripleItemType.values()) {
            if (value.getTripleItemType().equals(tripleItemType)) {
                return value;
            }
        }
        throw new IllegalArgumentException();
    }
}
